optax/_src/complex_valued.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019 DeepMind Technologies Limited. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Complex-valued optimization."""
```

Comment:
@mkunesch

Reply:
Ok, I'll wait for others' opinions about the namespace structure

Reply:
I'd like to name the module `complex_valued` rather than `complex`, because `complex` is a Python built-in and overriding it may cause unexpected results.